client/blocks/google-my-business-stats-nudge/index.js

```diff
@@ -32,11 +32,41 @@ class GoogleMyBusinessStatsNudge extends Component {
 		visible: PropTypes.bool,
 	};
 
+	state = {
+		siteSwitch: false,
+		siteId: null,
+	};
+
 	static defaultProps = {
 		visible: true,
 	};
 
 	componentDidMount() {
+		this.recordView();
+	}
+
+	static getDerivedStateFromProps( props, state ) {
+		if ( props.siteId !== state.siteId ) {
+			return {
+				siteId: props.siteId,
+				siteSwitch: !! ( state.siteId && props.siteId ),
+			};
+		}
+
+		if ( state.siteSwitch ) {
+			return {
+				siteSwitch: false,
+			};
+		}
+	}
+
+	componentDidUpdate() {
+		if ( this.state.siteSwitch ) {
```

Comment:
Why not check that directly here and drop `getDerivedStateFromProps` ?
```es6
componentDidUpdate( prevProps ) {
	if ( prevProps.siteId && this.props.siteId && this.props.siteId !=== prevProps.siteId ) {
		this.recordView();
	}
}
```

Reply:
Because `visible` might be `false` on site switch, we need to wait for the props to change.

Reply:
I see 👍 Could you please write a comment about it on `getDerivedStateFromProps`?